luigi_tools/task.py

```diff
@@ -282,6 +282,18 @@ def _warn_on_wrong_global_param_type(_self, global_param_name, global_param_valu
                     _warn_on_wrong_global_param_type, new_param
                 )
 
+                # Fix task_value() to not normalize when the parameter does not have default value
+                def task_value(_self, task_name, param_name):
+                    # pylint: disable=protected-access
+                    value = _self._get_value(task_name, param_name)
+                    if value == _no_default_value:
+                        return value
+                    else:
+                        return _self._actual_task_value(task_name, param_name)
+
+                new_param._actual_task_value = new_param.task_value
```

Comment:
I don't know if this statement is copying the new_param.task_value or just adding a reference to it. Just FYI.

Reply:
As far as I know it just adds a reference to the `task_value` method, which is what I need here so I can use it at line 292.

Reply:
it's ok as long as that's the desired behaviour.

Reply:
:+1: